subprojects/core-api/src/main/java/org/gradle/api/provider/HasMultipleValues.java

```diff
@@ -152,4 +143,34 @@ public interface HasMultipleValues<T> extends HasConfigurableValue {
      */
     @Override
     void finalizeValue();
+
+    /**
+     * Performs incremental updates to the actual value of this property.
+     *
+     * {@inheritDoc}
+     *
+     * For wholesale updates to the explicit value, use
+     * or {@link #set(Iterable)}, {@link #set(Provider)} .
+     *
+     * For wholesale updates to the convention value, use
+     * {@link #convention(Iterable)} or {@link #convention(Provider)}.
+     */
+    @Override
+    HasMultipleValues<T> withActualValue(Action<CollectionPropertyConfigurer<T>> action);
+
+    /**
+     * Performs incremental updates to the actual value of this property.
+     *
+     * This is a Groovy closure-compatible version of
+     * {@link HasMultipleValues#withActualValue(Action)},
+     * having this property's actual value (and not the property itself)
+     * as the target object.
+     *
+     * @param action a Groovy closure to incrementally configure this object's actual value
+     * via the {@link CollectionPropertyConfigurer} protocol.
+     *
+     * @see #withActualValue(Action)
+     * @since 8.7
+     */
+    HasMultipleValues<T> withActualValue(Closure<Void> action);
```

Comment:
:nail_care: Why `Closure` overload appears only on some types? Shouldn't it be declared in `ConfigurableValue`?

Reply:
I figured we could pull it up later if wanted without binary compatibility consequences - right?